### PR TITLE
[Hub Generated] Review request for Microsoft.Media to add version stable/2020-05-01

### PR DESCRIPTION
**Pull request by giakas**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/giakas/azure-rest-api-specs/dev-mediaservices-Microsoft.Media-2020-05-01...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by akning-ms** [comment]
@jianyexi  to take a look breaking change exception.

**Comment by jianyexi** [comment]
@giakas The breaking change exception is caused by the below value which is exceed the max Int64 , could you change it ? The max value for int64 is 9223372036854775807

https://github.com/Azure/azure-rest-api-specs/blob/cb2815447476460de069b90e26919e41c5f3e6ef/specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-list-all.json#L20

**Comment by jianyexi** [comment]
@giakas 
The breaking change exception is caused by the below value which is exceed the max Int64 in javascript which is used by autorest  .
could you decrease the value of "presentationWindowDuration" in below files , you can change it to 9223372036854774806 to address the exception.
in the below  examples:
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-create.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-list-all.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-update.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-create.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-get-by-name.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-list-all.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-update.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-create.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-list-all.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-update.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-create.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-get-by-name.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-list-all.json
specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-update.json

**Comment by giakas** [comment]
> @giakas
> The breaking change exception is caused by example values which is exceed the max Int64 in javascript which is used by autorest .
> could you decrease the value of "presentationWindowDuration" in below files , you can change it to 9223372036854774806 to address the exception in the below examples:
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-create.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-list-all.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/accountFilters-update.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-create.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-get-by-name.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-list-all.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2018-07-01/examples/assetFilters-update.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-create.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-list-all.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/accountFilters-update.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-create.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-get-by-name.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-list-all.json
> specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/examples/assetFilters-update.json

@jianyexi  I updated them with 9223372036854774806  but it still failing.

File: specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/Accounts.json [hunk 1]
```diff
@@ -159,6 +159,35 @@
           "readOnly": true,
           "x-nullable": false
         },
+        "lockAggregationType": {
```
Comment (jorgecotillo) [inline specification/mediaservices/resource-manager/Microsoft.Media/stable/2020-05-01/Accounts.json:162]: >lockAggregationType [](start = 9, length = 19)

Just to make sure, this value is set by your service right?

**Review by allenjzhang — APPROVED** [review]

**Comment by johndeu** [comment]
@allenjzhang @iscai-msft - How much longer to get the Node.Js SDK updated?

**Comment by iscai-msft** [comment]
@johndeu what do you mean by the Node.Js SDK, do you mean the typescript SDK generator?

**Comment by johndeu** [comment]
Yes.  The typescript library in npm.

Thanks,
John D

________________________________

**Comment by iscai-msft** [comment]
forwarding to @joheredi he works on the typescript library

**Pull request merged**